src/mbgl/text/placement.cpp

```diff
@@ -1467,13 +1476,22 @@ bool TilePlacement::canPlaceAtVariableAnchor(const CollisionBox& box,
                                              const mat4& posMatrix,
                                              float textPixelRatio) {
     assert(tileBorders);
+    if (populateIntersections) {
+        // A variable label is only allowed to intersect tile border with the first anchor.
+        if (anchor == anchors.front()) {
```

Comment:
What if the first anchor is "left" with an icon on the right side and icon intersects right tile border? I think icon might be cut-off in that scenario if right tile has higher priority label in that position.

Reply:
Discussed offline, icon is always placed at anchor position. Check for text collision box is sufficient.